heron/proto/ckptmgr.proto

```diff
@@ -0,0 +1,182 @@
+package heron.proto.ckptmgr;
+option java_package = "com.twitter.heron.proto.ckptmgr";
+option java_outer_classname = "CheckpointManager";
+
+import "common.proto";
+import "physical_plan.proto";
+
+// message stored in zk by the tmaster
+message StatefulConsistentCheckpoints {
+  // This is the most recent checkpoint that we believe exists
+  required string most_recent_checkpoint_id = 1;
```

Comment:
Instead of storing an string id and a string[] backupId, could we instead store an ordered array of Checkpoint objects, most recent first? Using an object instead of a string would allow for extensibility to bind more metadata to a checkpoint than just id (e.g., creation time, location).  We could adjust how many checkpoints we keep via config (either time based or absolute count), which becomes the size of the array.